src/Operations.jl

```diff
@@ -941,9 +944,18 @@ function build_versions(ctx::Context, uuids::Set{UUID}; verbose=false)
         pkg = PackageSpec(;uuid=uuid, name=name, version=version)
         build_file = buildfile(source_path)
         # compatibility shim
-        build_project_override = isfile(projectfile_path(builddir(source_path))) ?
-            nothing :
-            gen_target_project(ctx.env, ctx.registries, pkg, source_path, "build")
+        local build_project_override, build_project_preferences
+        if isfile(projectfile_path(builddir(source_path)))
+            build_project_override = nothing
+            Pkg.activate(projectfile_path(builddir(source_path))) do
+                build_project_preferences = Base.get_preferences()
+            end
```

Comment:
@KristofferC is it okay for us to use `Pkg.activate()` here, or should we manually just set/reset `Base.LOAD_PATH` like I do in [these tests here](https://github.com/JuliaLang/julia/blob/7a8613f1c1aac6eda0e6b7e93c24d67bdb187a5a/test/loading.jl#L878-L880)?  I'm not sure if there's extra work that `Pkg.activate()` is doing beyond just manipulating `LOAD_PATH`.

Reply:
It is invalid for non-default load paths.

Reply:
What is invalid?

Reply:
`Pkg.activate`

Reply:
Alright, we'll do the `Base.LOAD_PATH` mcjigggery then